### PR TITLE
Use upstream imgur library

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Imgur media upload

### Description
This PR replaces the dependency `node-imgur`.
We use a hackmd fork of "node-imgur" for no reason. The upstream lib is way more advanced and modern.

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Review by DerMolly — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Comment by davidmehren** [comment]
It might be a better idea to backport https://github.com/hedgedoc/hedgedoc/blob/develop/src/media/backends/imgur-backend.ts which just uses `node-fetch`, as that saves us from adding a pile of new dependencies.

**Comment by DerMolly** [comment]
> It might be a better idea to backport [https://github.com/hedgedoc/hedgedoc/blob/develop/src/media/backends/imgur-backend.ts](https://github.com/hedgedoc/hedgedoc/blob/develop/src/media/backends/imgur-backend.ts?rgh-link-date=2021-04-15T20%3A21%3A33Z) which just uses `node-fetch`, as that saves us from adding a pile of new dependencies.

I just did that.

File: lib/web/imageRouter/imgur.js [hunk 1]
```diff
@@ -15,12 +15,25 @@ exports.uploadImage = function (imagePath, callback) {
...
-    .then(function (json) {
+  const buffer = fs.readFileSync(imagePath)
+
+  const params = new URLSearchParams()
+  params.append('image', buffer.toString('base64'))
+  params.append('type', 'base64')
+  fetch('https://api.imgur.com/3/image', {
+    method: 'POST',
+    body: params,
+    headers: { Authorization: `Client-ID ${config.imgur.clientID}` }
```
Comment (mrdrogdrog) [inline lib/web/imageRouter/imgur.js:26]: @DerMolly 

What if you don't provide an client id? The imgur lib defined a fallback id for this purpose. Maybe we should also fallback to it.

Comment (DerMolly) [reply lib/web/imageRouter/imgur.js:26]: @SISheogorath Does the current imgur support uploads without configuring a client id?

Comment (DerMolly) [reply lib/web/imageRouter/imgur.js:26]: If so we should probably continue to do so…

Comment (mrdrogdrog) [reply lib/web/imageRouter/imgur.js:26]: If you inspect the code.. yes

setClientID accepts only strings. If you don't set the ID in your config, then the lib will continue to use the build-in ID.

Comment (mrdrogdrog) [reply lib/web/imageRouter/imgur.js:26]: https://github.com/hackmdio/node-imgur/blob/de0a7a1f1eb2cb6628385fedb990ad396a190573/lib/imgur.js#L285

Comment (DerMolly) [reply lib/web/imageRouter/imgur.js:26]: Do we want to continue to use that ID or do we want our own? And do we want to continue this behavior in 2.0 (currently it fails with no clientId)?

Comment (davidmehren) [reply lib/web/imageRouter/imgur.js:26]: I think without a client ID, deletion of uploads is impossible?
If so, we should not allow an empty client id.

Comment (mrdrogdrog) [reply lib/web/imageRouter/imgur.js:26]: To require a client id now would be a breaking change. Therefore we should keep a fallback id for now for 1.x

Comment (DerMolly) [reply lib/web/imageRouter/imgur.js]: I restored the default behavior for this PR

File: lib/web/imageRouter/imgur.js [hunk 2]
```diff
@@ -15,12 +15,27 @@ exports.uploadImage = function (imagePath, callback) {
     return
   }
 
-  imgur.setClientId(config.imgur.clientID)
-  imgur.uploadFile(imagePath)
-    .then(function (json) {
+  const clientId = config.imgur.clientID || 'f0ea04148a54268'
```
Comment (mrdrogdrog) [inline lib/web/imageRouter/imgur.js:18]: @DerMolly 
Use 032aa2f687790cd. But please add a comment.

Comment (DerMolly) [reply lib/web/imageRouter/imgur.js]: Done

**Review by InnayTool — DISMISSED** [review]

**Review by davidmehren — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Pull request merged**